src/test/fuzz/script_flags.cpp

```diff
@@ -15,7 +16,75 @@
 #include <utility>
 #include <vector>
 
-FUZZ_TARGET(script_flags)
+namespace {
+
+inline uint64_t HashSig(Span<const unsigned char> sig)
+{
+    CSipHasher hasher{0xdead, 0xbeef};
+    hasher.Write(sig);
+    return hasher.Finalize();
+}
+// Reduce a CScriptNum to one bit
+inline bool HashScriptNum(const CScriptNum& num)
+{
+    CSipHasher hasher{0xdead, 0xbeef};
+    hasher.Write(num.getvch());
+    return hasher.Finalize() & 1;
+}
+
+class FuzzedSignatureChecker : public BaseSignatureChecker
+{
+public:
+    FuzzedSignatureChecker(const CTransaction* tx, unsigned int in,
+                           const CAmount& amount, const PrecomputedTransactionData& tx_data,
+                           MissingDataBehavior mdb) {}
+
+    bool CheckECDSASignature(const std::vector<unsigned char>& sig, const std::vector<unsigned char>& pub_key,
+                             const CScript& script_code, SigVersion sig_version) const override
+    {
+        return !sig.empty() && (HashSig(sig) & 1);
+    }
+    bool CheckSchnorrSignature(Span<const unsigned char> sig, Span<const unsigned char> pub_key,
+                               SigVersion sig_version, ScriptExecutionData& exec_data,
+                               ScriptError* script_error = nullptr) const override
+    {
+        uint64_t fuzz_hash{HashSig(sig)};
```

Comment:
What does hashing adds here? If you need more than one bit why not take it out of `sig` directly (and have a default value if it's too small)?